tests/framework/matcher/conditions.go

```diff
@@ -0,0 +1,180 @@
+package matcher
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/onsi/gomega/format"
+	"github.com/onsi/gomega/types"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"kubevirt.io/kubevirt/tests/framework/matcher/helper"
+)
+
+func ConditionMissingOrFalse(conditionType interface{}) types.GomegaMatcher {
+	return conditionMatcher{
+		expectedType:          conditionType,
+		expectedStatus:        k8sv1.ConditionFalse,
+		conditionCanBeMissing: true,
+	}
+}
+
+func ConditionTrue(conditionType interface{}) types.GomegaMatcher {
+	return conditionMatcher{
+		expectedType:          conditionType,
+		expectedStatus:        k8sv1.ConditionTrue,
+		conditionCanBeMissing: false,
+	}
+}
+
+type conditionMatcher struct {
+	expectedType          interface{}
+	expectedStatus        k8sv1.ConditionStatus
+	conditionCanBeMissing bool
+}
+
+func (c conditionMatcher) Match(actual interface{}) (success bool, err error) {
+	if helper.IsNil(actual) {
```

Comment:
Maybe use the `ToUnstructured` here. It coveres also spcial cases around pointers. It would also return an error if `IsNil` to not by accident match a negation on the matcher.

You can check https://github.com/kubevirt/kubevirt/blob/296612a815b64eb5bceefa4d53a686cc124d83ec/tests/framework/matcher/owner.go#L23.

For not supporting slices you can still keep the IsSlice check.